clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -974,21 +969,6 @@ private void handleInitiateApiVersionRequests(long now) {
         }
     }
 
-    /**
-     * Validate that the response corresponds to the request we expect or else explode
-     */
-    private static void correlate(RequestHeader requestHeader, ResponseHeader responseHeader) {
```

Comment:
nit: I don't think we need to remove this helper

Reply:
Why do you want to keep it if it is not used?